app/models/user.rb

```diff
@@ -926,7 +926,7 @@ def can_update_competition_series?(competition)
   end
 
   def can_upload_competition_results?(competition)
-    can_submit_competition_results?(competition, upload_only: true)
+    can_admin_results? || can_submit_competition_results?(competition, upload_only: true)
```

Comment:
Why is this change becoming necessary now?

Reply:
In particular, I don't see any other occurence of `can_upload_competition_results?` in your whole PR

Reply:
The `upload_json` method already has a validation that it will work only if `can_upload_competition_results?` is `true`. Earlier, this check was only for Delegates and because of that there were additional checks like whether results are already posted, whether competition is in_progress/over, whether competition is announced, etc. But these are not applicable when WRT is uploading. They should have more power and should be able to do it anytime.

Reply:
Can you please point me to the exact lines of code where the pre-existing checks that you mention are defined?
I am very wary about updating existing permission methods and want to make 101% sure this doesn't break anything

Reply:
https://github.com/thewca/worldcubeassociation.org/blob/8d5ac7d7397dd3624b81fdf66e74b11f0bdd80c8/app/controllers/results_submission_controller.rb#L7

Reply:
In that case I think you should adjust `can_submit_competition_results?` instead.
The method `can_upload_competition_results?` only serves as a "convenience alias" to `can_submit` with the optional `upload_only` parameter set to false. All of the core logic happens in `can_submit`, including an already existing check of `can_admin_results?` at some point.

Reply:
"complicated" is a relative term. More lines of code, yes. But also more _meaningful_ lines of code by far.

I think the naming of variables as well as the final return value in `can_submit` do a very good job at explaining what's going on, and they are fairly readable. The fact that the code now gets even more bloated is not your fault, it's the fault of the real-world WCA logic which just happens to be complicated.